autospec/tarball.py

```diff
@@ -573,18 +651,22 @@ def process(url_arg, name_arg, ver_arg, target, archives_arg, filemanager):
     version = ver_arg
     archives = archives_arg
     tarfile = os.path.basename(url_arg)
-    # determine name and version of package
-    name_and_version(name_arg, ver_arg, filemanager)
     # determine build pattern and build requirements from url
     detect_build_from_url(url)
+    # Create the download path for content and set build.download_path
+    create_download_path(target)
+    # determine name and version of package
+    name, rawname, version = name_and_version(name_arg, ver_arg, filemanager)
+    # '-' aren't allowed in the version string for spec files
+    # rpmbuild will fail so replace them with '.'
+    if '-' in version:
```

Comment:
Can you move this version mangling code into a separate function? Maybe into util.py?